src/pubsub.rs

```diff
@@ -139,7 +139,28 @@ impl PubSub {
     where
         S: Into<String>,
     {
-        self.subscriptions.insert(topic.into());
+        let topic_str = topic.into();
+        if self.subscriptions.contains(&topic_str) || self.connection.is_none() {
+            println!("already subscribed to topic or not connected");
```

Comment:
No need to be verbose here, could just return early. Caller may not want to print a message